android-sdk/src/main/java/com/mobilecoin/lib/MobileCoinClient.java

```diff
@@ -1219,18 +1237,27 @@ List<Ring> getRingsForUTXOs(
             @NonNull UnsignedLong numTxOutsInLedger,
             @NonNull Rng rng
     ) throws InvalidFogResponse, NetworkException, AttestationException {
+        Logger.i(TAG, "Getting rings for utxos", null,
+                "utxo count:", utxos.size(),
+                "num txos in ledger", numTxOutsInLedger
+        );
         // Sanity check to ensure all UTXOs have unique indices
         HashSet<UnsignedLong> indices = new HashSet<>();
         for (OwnedTxOut utxo : utxos) {
             if (!indices.add(utxo.getTxOutGlobalIndex())) {
-                throw new IllegalStateException("utxos contains non-unique indices");
+                final IllegalStateException ise = new IllegalStateException("utxos contains non-unique indices");
+                Logger.e(TAG, "Encountered Exception while getting rings", ise);
```

Comment:
Using "Encoutered" for the exception here seems strange since this is the throwing code (later code that catches exceptions also uses "Encountered" for the caught exceptions).  Maybe "Throwing Exception - error while getting rings", ise) or something like that for the all the throwing locations in this file?

Reply:
Hmm, yeah, this is an excellent point. In the logs it could look like two Exceptions occurred. I took your suggestion and re-worded it to say "throwing" here and in a couple other places.

Great find!